### ISSUE TITLE
Copying data from grid takes a very long time for moderately sized data extracts

### ISSUE THREAD

**Issue by tatemian** [microsoft/azuredatastudio#25284]
Type: <b>Performance Issue</b>

Pretty straightforward. I work with large query returns, often in excess of 100,000 rows. Copying data from the data grid takes a _very_ long time (well in excess of an hour, most often) to copy data to the clipboard, even for small data returns. There is just a small popup in the bottom right that says "Copying..." and there's no way to cancel the action other than running the query again or closing it. Of course, neither help the time it takes copy the data grid.

I'm rather new to ADS, but I don't remember it being like this when I first started using it. I'm currently having to save the data grid to a temp .csv file and copy the data from another program like Excel. Just a slight inconvenience, but it adds up with countless queries throughout the day.

Updated to 1.47.1 today and that did not solve the issue.

Any help is greatly appreciated.

Azure Data Studio version: azuredatastudio 1.47.1 (b6f7beb01f92adaa4b79b6b6f3ac704e95cafe6e, 2024-01-10T02:15:02.628Z)
OS version: Windows_NT x64 10.0.19045
Restricted Mode: No
Preview Features: Enabled
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz (8 x 2112)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|7.82GB (0.71GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Workspace Info</summary>

```
;
```

</details>
Extensions: none

**Comment by bwyatt-mcr** [comment]
Yep. My issue #25282 is related to this, but my data sets are far smaller and it still struggles to do these basic tasks.

**Comment by kisantia** [comment]
There was a recent change made to speed this up: https://github.com/microsoft/azuredatastudio/pull/25152. Can you try [insiders](https://github.com/microsoft/azuredatastudio?tab=readme-ov-file#try-out-the-latest-insiders-build-from-main-branch) and see if the performance has improved?

**Comment by tatemian** [comment]
> There was a recent change made to speed this up: #25152. Can you try [insiders](https://github.com/microsoft/azuredatastudio?tab=readme-ov-file#try-out-the-latest-insiders-build-from-main-branch) and see if the performance has improved?

I tried using v.1.48.0-insider. No notable improvement to grid copy speed. Referenced improvement simulated 50,000 rows of 4 columns resulting in ~3 second copy. I recreated that scenario in insiders and attempted grid copy of the data, which took 5 min, 12 seconds. I attempted a grid copy of 150,000 records with 30 columns, but cancelled after several hours of waiting.

**Comment by cheenamalhotra** [comment]
Can you give an example of what datatypes you're copying here?
That would help us extrapolate data and reproduce the issue..

**Comment by tatemian** [comment]
> Can you give an example of what datatypes you're copying here? Sample data would be great too! That would help us extrapolate data and reproduce the issue..

Unfortunately, I cannot supply sample data due to the nature of my work, but here are the data types for the fields I used in the above test:

1. [int, not null]
2. [datetime, null]
3. [char(1), null]
4. [char(14), null]
 
5. [varchar(64), null]
6. [varchar(64), null]
7. [varchar(64), null]
8. [char(25), null]

**Comment by serhat359** [comment]
Version 1.50.0 seems to have improved the performance of copying to clipboard. This issue might be resolved.
